lib/store/podDataSlice.ts

```diff
@@ -0,0 +1,51 @@
+import { createSlice, PayloadAction } from "@reduxjs/toolkit";
+import {
+    createPodDataFromAdapter,
+    PacketUpdate,
+    PodDataAdapter,
+} from "../adapters";
+import { PodData, updatePodData as updatePackets, Board } from "../models";
+import { create, StateCreator, StoreApi, UseBoundStore } from "zustand";
+
+export interface PodDataSlice {
+    podData: PodData
+    initPodData: (podDataAdapter: PodDataAdapter) => void
+    updatePodData: (newPodData: Record<number, PacketUpdate>) => void
+}
+
+export const podDataSlice: StateCreator<PodDataSlice> = (set, get) => ({
+    podData: {
+        boards: [] as Board[],
+        packetToBoard: {} as Record<number, number>,
+        lastUpdates: {} as Record<string, PacketUpdate>,
+    },
+
+    /**
+     * Reducer that initializes the state based on podDataAdapter.
+     * It uses a helper function createPodDataFromAdapter to do it.
+     * @param {PodDataAdapter} podDataAdapter 
+     */
+    initPodData: (podDataAdapter: PodDataAdapter) => {
+        set(state => ({
+            ...state,
+            podData: createPodDataFromAdapter(podDataAdapter)
+        }))
+    },
+
+    /**
+     * Reducer that updates the state based on newPodData.
+     * @param {Record<number, PacketUpdate>} newPodData 
+     */
+    updatePodData: (newPodData: Record<number, PacketUpdate>) => {
+
+        set(state => ({
+            ...state,
+            podData: {
+                ...state.podData,
+                lastUpdates: newPodData
+            }
+        }))
+
+        updatePackets(get().podData, newPodData) // TODO: CHECK THIS. IM PRETTY SURE IT DOESN'T WORK
```

Comment:
This is just not good, as the comment says :)
This logic just updates the pod data afterwards by modifying the object, I would change it to do such logic in this method in this file, have it loop through the updates and change the values in the store `updatePodData` method.

I recommend you make a new issue for this change as it has a substantial amount of code involved and it would make this PR too big.